Tests/QuaternionTests/PropertyTests.swift

```diff
@@ -100,13 +100,80 @@ final class PropertyTests: XCTestCase {
       XCTAssertEqual(infs[0], i)
       XCTAssertEqual(infs[0].hashValue, i.hashValue)
     }
+    // Validate that all *normal* values hash their absolute components, so
+    // that rotations in *R³* of `q` and `-q` will hash to same value.
+    let pairs: [(lhs: Quaternion<T>, rhs: Quaternion<T>)] = [
+      (
+        Quaternion<T>(real:  .pi, imaginary:  .pi,  .pi,  .pi),
+        Quaternion<T>(real: -.pi, imaginary:  .pi,  .pi,  .pi)
```

Comment:
These should not hash to the same value; the pairs should be `-pi, pi, pi, pi` and `pi, -pi, -pi, -pi`, right?

Reply:
No, that's not correct - my bad!
The `canonicalTransform` implementation had a bug and these tests are wrong too.

Bug fixed and tests adjusted - thank you!